src/eve-server/system/DestinyManager.cpp

```diff
@@ -3068,21 +3204,63 @@ void DestinyManager::SendTerminalExplosion(uint32 shipID, uint32 bubbleID, bool
 }
 
 void DestinyManager::SendSetState() const {
-    if (!mySE->HasPilot())
+    if (!mySE->HasPilot()) {
         return;
+    }
 
-    if (is_log_enabled(DESTINY__MESSAGE))
-        _log(DESTINY__MESSAGE, "Destiny::SendSetState() Called for Ship:%s(%u) Pilot:%s(%u)", \
-                        mySE->GetName(), mySE->GetID(), mySE->GetPilot()->GetName(), mySE->GetPilot()->GetCharacterID());
+    if (is_log_enabled(DESTINY__MESSAGE)) {
+        _log(
+            DESTINY__MESSAGE,
+            "Destiny::SendSetState() Called for Ship:%s(%u) Pilot:%s(%u)",
+            mySE->GetName(),
+            mySE->GetID(),
+            mySE->GetPilot()->GetName(),
+            mySE->GetPilot()->GetCharacterID()
+        );
+    }
+
+    // if the player is not warping, tell the client they're not warping.
+    // As of 2024-10-03, this doesn't always work and there are still issues
+    // with the client sometimes not starting a warp sequence.
+    std::vector<PyTuple*> updates;
+    OnSpecialFX10 sfx;
+    sfx.guid = "effects.Warping";
+    sfx.entityID = mySE->GetID();
+    sfx.isOffensive = false;
+    sfx.start = false;
+    sfx.active = false;
+    if (m_ballMode == Destiny::Ball::Mode::WARP) {
+        sfx.start = true; // TODO: verify if this is necessary
+        sfx.active = true;
+    }
+
+    updates.push_back(sfx.Encode());
+    SendDestinyUpdate(updates);
+    updates.clear();
 
     SetState ss;
-        ss.stamp = sEntityList.GetStamp();
-        ss.ego = mySE->GetID();
+
+    ss.stamp = sEntityList.GetStamp();
+    ss.ego = mySE->GetID();
+
+    if (mySE->SysBubble() == nullptr) {
+        // returning here preemptively avoids a segfault (good), but isn't a
+        // good situation to encounter.
+        sLog.Error(
+            "DestinyManager::SendSetState()",
+            "Destiny::SendSetState() the player isn't in a system bubble! Aborting attempt to send state."
+        );
+
+        return;
+    }
```

Comment:
Updating the sendstate function here for two reasons:

1. (unsuccessfully) attempting to fix the 'warp drive active' client/server desync issue that sometimes happens. This issue really bothers me but I still haven't figured it out.
2. gracefully avoiding the segfault that occurs when attempting to sendstate while having a `nullptr` sysbubble.